From: "roberto@quantal64" <roberto@quantal64>
Date: Sat Jul  7 10:32:21 2012 +0200
Subject: skip headers sending when no headers are defined
Origin: http://projects.unbit.it/hg/uwsgi-1.2/rev/740360f10046

# HG changeset patch
# User roberto@quantal64
# Date 1341649941 -7200
# Node ID 740360f100465e187bf6bd70c9c85a4c1ad81e43
# Parent  6e378241b7b777143951a4db24190bd1af801fdd
skip headers sending when no headers are defined

diff -r 6e378241b7b7 -r 740360f10046 plugins/python/wsgi_headers.c
--- a/plugins/python/wsgi_headers.c	Sat Jul 07 10:31:28 2012 +0200
+++ b/plugins/python/wsgi_headers.c	Sat Jul 07 10:32:21 2012 +0200
@@ -253,6 +253,8 @@
 
 int uwsgi_python_do_send_headers(struct wsgi_request *wsgi_req) {
 
+	if (!wsgi_req->headers_hvec) return 0;
+
 #ifdef __sun__
         int remains = wsgi_req->headers_hvec + 1;
         int iov_size;
